velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -676,85 +567,25 @@ void HiveDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
 
-  reader_ = dwio::common::getReaderFactory(readerOpts_.getFileFormat())
-                ->createReader(std::move(input), readerOpts_);
+  auto fileHandle = fileHandleFactory_->generate(split_->filePath).second;
+  auto input = createBufferedInput(*fileHandle, readerOpts_);
 
-  emptySplit_ = false;
-  if (reader_->numberOfRows() == 0) {
-    emptySplit_ = true;
-    return;
-  }
-
-  // Check filters and see if the whole split can be skipped.
-  if (!testFilters(
-          scanSpec_.get(),
-          reader_.get(),
-          split_->filePath,
-          split_->partitionKeys,
-          partitionKeys_)) {
-    emptySplit_ = true;
-    ++runtimeStats_.skippedSplits;
-    runtimeStats_.skippedSplitBytes += split_->length;
-    return;
-  }
-
-  auto& fileType = reader_->rowType();
-  // Keep track of schema types for columns in file, used by ColumnSelector.
-  std::vector<TypePtr> columnTypes = fileType->children();
-
-  auto& childrenSpecs = scanSpec_->children();
-  for (size_t i = 0; i < childrenSpecs.size(); ++i) {
-    auto* childSpec = childrenSpecs[i].get();
-    const std::string& fieldName = childSpec->fieldName();
-
-    auto iter = split_->partitionKeys.find(fieldName);
-    if (iter != split_->partitionKeys.end()) {
-      setPartitionValue(childSpec, fieldName, iter->second);
-    } else if (fieldName == kPath) {
-      setConstantValue(childSpec, VARCHAR(), velox::variant(split_->filePath));
-    } else if (fieldName == kBucket) {
-      if (split_->tableBucketNumber.has_value()) {
-        setConstantValue(
-            childSpec,
-            INTEGER(),
-            velox::variant(split_->tableBucketNumber.value()));
-      }
-    } else {
-      auto fileTypeIdx = fileType->getChildIdxIfExists(fieldName);
-      if (!fileTypeIdx.has_value()) {
-        // Column is missing. Most likely due to schema evolution.
-        VELOX_CHECK(readerOpts_.getFileSchema());
-        setNullConstantValue(
-            childSpec, readerOpts_.getFileSchema()->findChild(fieldName));
-      } else {
-        // Column no longer missing, reset constant value set on the spec.
-        childSpec->setConstantValue(nullptr);
-        auto outputTypeIdx = readerOutputType_->getChildIdxIfExists(fieldName);
-        if (outputTypeIdx.has_value()) {
-          // We know the fieldName exists in the file, make the type at that
-          // position match what we expect in the output.
-          columnTypes[fileTypeIdx.value()] =
-              readerOutputType_->childAt(*outputTypeIdx);
-        }
-      }
-    }
-  }
-
-  scanSpec_->resetCachedValues(false);
-  configureRowReaderOptions(
-      rowReaderOpts_,
-      ROW(std::vector<std::string>(fileType->names()), std::move(columnTypes)));
-  // NOTE: we firstly reset the finished 'rowReader_' of previous split before
-  // setting up for the next one to avoid doubling the peak memory usage.
-  rowReader_.reset();
-  rowReader_ = createRowReader(rowReaderOpts_);
+  splitReader_ = SplitReader::create(
```

Comment:
`splitReader_.reset()` first since that will avoid doubling the peak memory usage.
Recently, `baseRowReader's` peak memory usage was fixed in this way.